src/overlays/actors/ovl_En_Wonder_Talk2/z_en_wonder_talk2.c

```diff
@@ -26,19 +31,274 @@ const ActorInit En_Wonder_Talk2_InitVars = {
+// u32 D_80B3A8E0[] = {0x00060000, 0x00010002, 0x00030004, 0x00050000};
+
+s16 D_80B3A8E0[] = { 0x6, 0x0, 0x1, 0x2, 0x3, 0x4, 0x5 };
+
+void EnWonderTalk2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void EnWonderTalk2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnWonderTalk2* this = THIS;
+    s32 offsetCounter;
+    s16 zOffset;
+
+    osSyncPrintf("\n\n");
+    osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ 透明メッセージ君 ☆☆☆☆☆ %x\n" VT_RST, this->actor.params);
+    this->unk_150 = (this->actor.params >> 6) & 0xFF;
+    if (this->actor.posRot.rot.z > 0) {
+        offsetCounter = 0;
+        zOffset = this->actor.posRot.rot.z;
+
+        while (zOffset > 10) {
+            zOffset -= 10;
+            offsetCounter += 1;
+        }
+
+        this->unk_15C = zOffset * 40.0f;
+        if (offsetCounter > 6) {
+            offsetCounter = 0;
+        }
+
+        this->actor.unk_1F = D_80B3A8E0[offsetCounter];
+
+        osSyncPrintf("\n\n");
+        osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ 元は？       ☆☆☆☆☆ %d\n" VT_RST, this->actor.posRot.rot.z);
+        osSyncPrintf(VT_FGCOL(PURPLE) "☆☆☆☆☆ レンジは？   ☆☆☆☆☆ %d\n" VT_RST, this->actor.unk_1F);
+        osSyncPrintf(VT_FGCOL(CYAN) "☆☆☆☆☆ は、範囲わ？ ☆☆☆☆☆ %f\n" VT_RST, this->unk_15C);
+        osSyncPrintf("\n\n");
+        osSyncPrintf("\n\n");
+        osSyncPrintf("\n\n");
+    }
+    this->posRot = this->actor.posRot.pos;
+    this->switchFlag = (this->actor.params & 0x3F);
+    this->unk_154 = ((this->actor.params >> 0xE) & 3);
+
+    if (this->switchFlag == 0x3F) {
+        this->switchFlag = -1;
+    }
+    if (this->switchFlag >= 0 && Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ Ｙｏｕ ａｒｅ Ｓｈｏｃｋ！  ☆☆☆☆☆ %d\n" VT_RST, this->switchFlag);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    if ((this->unk_154 == 1) && (globalCtx->sceneNum == SCENE_MEN) && (this->switchFlag != 8) &&
+        (this->switchFlag != 0x16) && (this->switchFlag != 0x2F)) {
+
+        this->unk_15A = 0;
+        this->unk_154 = 4;
+    }
+    if (this->unk_154 == 3) {
+        this->actor.flags &= 0xF7FFFFFF;
+        this->actionFunc = EnWonderTalk2_DoNothing;
+    } else {
+        this->actionFunc = func_80B3A10C;
+    }
+}
+
+void func_80B3A10C(EnWonderTalk2* this, GlobalContext* globalCtx) {
+    this->actor.textId = 0x200;
+    this->actor.textId |= this->unk_150;
+    if (this->unk_154 == 1 || this->unk_154 == 4) {
+        this->actionFunc = func_80B3A4F8;
+    } else {
+        this->actionFunc = func_80B3A15C;
+    }
+}
+
+void func_80B3A15C(EnWonderTalk2* this, GlobalContext* globalCtx) {
+    Player* player;
+    Actor* actor;
+    s16 yawDiff;
+    s16 yawDiffTemp;
+
+    actor = &this->actor;
+    player = PLAYER;
+    this->unk_158++;
+    if (this->switchFlag >= 0 && Flags_GetSwitch(globalCtx, this->switchFlag)) {
+        if (this->unk_15A == 0) {
+            this->actor.flags &= -2;
+            this->unk_15A = 1;
+            return;
```

Comment:
This `return` should be unnecessary since the rest of the function would never be reached anyway